File: hashmap/383-ransom-note/RansomNote.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character, Integer> count = new HashMap<>();
        for(char ch:magazine.toCharArray()){
            count.put(ch, count.getOrDefault(ch, 0)+1);
        }
        for(char ch:ransomNote.toCharArray()){
            if(!count.containsKey(ch) || count.get(ch)==0){
                return false;
            }
            count.put(ch, count.get(ch)-1);
        }
        return true;
    }
}

File: hashmap/383-ransom-note/RansomNote.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        count ={}
        for ch in magazine:
            count[ch] = count.get(ch, 0)+1
        for c in ransomNote:
            if c not in count or count[c]==0:
                return False
            count[c]-=1
        return True